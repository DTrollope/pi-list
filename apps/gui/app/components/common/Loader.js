import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const propTypes = {
    className: PropTypes.string,
    size: PropTypes.string
};

const defaultProps = {
    className: '',
    size: 'default',
};

const Loader = (props) => {
    const spinnerClassNames = classNames(
        'lst-loader',
        props.className,
        {
            // Loader sizes
            'lst-loader--small': props.size === 'small',
            'lst-loader--mini': props.size === 'mini',
            'lst-loader--large': props.size === 'large',

            'lst-loader--inverted': props.inverted === true
        },
    );

    return (
        <span className={spinnerClassNames} />
    );
};

Loader.propTypes = propTypes;
Loader.defaultProps = defaultProps;

export default Loader;
