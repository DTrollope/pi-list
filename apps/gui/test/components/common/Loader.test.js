import React from 'react';
import { shallow } from 'enzyme';
import Loader from 'components/common/Loader';
import renderer from 'react-test-renderer';

describe('components/common/Loader', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <Loader text="Hey! It's a test"/>
        );
    });

    describe('Loader Component', () => {
        it('should include a custom className', () => {
            wrapper.setProps({ className: 'custom-classname' });

            expect(wrapper.find('.lst-loader').hasClass('custom-classname')).toBe(true);
        });
    });

    describe('Loader Snapshots', () => {
        it('should render a default loader', () => {
            const tree = renderer
                .create(<Loader />)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a mini loader', () => {
            const tree = renderer
                .create(<Loader size="mini"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a small loader', () => {
            const tree = renderer
                .create(<Loader size="small"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });

        it('should render a large loader', () => {
            const tree = renderer
                .create(<Loader size="large"/>)
                .toJSON();

            expect(tree).toMatchSnapshot();
        });
    });
});
