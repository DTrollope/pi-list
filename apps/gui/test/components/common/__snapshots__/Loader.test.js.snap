// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/common/Loader Loader Snapshots should render a default loader 1`] = `
<span
  className="lst-loader"
/>
`;

exports[`components/common/Loader Loader Snapshots should render a large loader 1`] = `
<span
  className="lst-loader lst-loader--large"
/>
`;

exports[`components/common/Loader Loader Snapshots should render a mini loader 1`] = `
<span
  className="lst-loader lst-loader--mini"
/>
`;

exports[`components/common/Loader Loader Snapshots should render a small loader 1`] = `
<span
  className="lst-loader lst-loader--small"
/>
`;
